noxfile.py

```diff
@@ -0,0 +1,84 @@
+import os
+from pathlib import Path
+
+import nox
+
+LINT_FILES = ("hacking/pr_labeler/label.py", "noxfile.py")
+PINNED = os.environ.get("PINNED", "true").lower() in {"1", "true"}
+nox.options.sessions = ("lint",)
+
+
+def install(session: nox.Session, *args, req: str, **kwargs):
+    arguments = ["-r", f"tests/{req}.in"]
+    if PINNED:
+        arguments.extend(("-c", f"tests/{req}.txt"))
+    session.install(
+        *arguments,
+        *args,
+        **kwargs,
+    )
+
+
+@nox.session
+def static(session: nox.Session):
+    """
+    Run static checkers
+    """
+    install(session, req="static")
+    session.run("ruff", *session.posargs, *LINT_FILES)
+
+
+@nox.session
+def formatters(session: nox.Session):
+    """
+    Reformat code
+    """
+    install(session, req="formatters")
+    session.run("isort", *session.posargs, *LINT_FILES)
+    session.run("black", *session.posargs, *LINT_FILES)
+
+
+@nox.session
+def formatters_check(session: nox.Session):
+    """
+    Check code formatting without making changes
+    """
+    install(session, req="formatters")
+    session.run("isort", "--check", *session.posargs, *LINT_FILES)
+    session.run("black", "--check", *session.posargs, *LINT_FILES)
+
+
+@nox.session
+def typing(session: nox.Session):
+    install(session, req="typing")
+    session.run("mypy", *session.posargs, *LINT_FILES)
```

Comment:
It'd be nice to have multiple type checkers integrated, since different editors use their own implementations that don't actually run `mypy` most of the time…

Reply:
We *could* integrate pyright or pyre, but I'm not super interested. mypy is the defacto standard and integrating redundant checkers creates a mess of `noqa` comments when you need to disable something.

Reply:
Don't they all process standard type ignores?

Reply:
I think pyre has its own pyre-ignore syntax.

Reply:
I'll mark this as resolved, as I'm not interest in handling that in this PR.

Reply:
Well, I think pyright would be of most interest, since it's a de-facto standard in vscode, considering its popularity.